src/checkstyle/checks.xml

```diff
@@ -32,6 +32,10 @@
         <property name="message" value="Blank line before end of file" />
     </module>
     <module name="RegexpSingleline">
+        <property name="format" value="(class|interface) ([a-zA-Z0-9])* (extends|implements)" />
```

Comment:
`(class|interface) ([a-zA-Z0-9])*(|(<.*>))(\s)*(extends|implements)` 0 this will match generics of any depth.